code-studio/src/js/components/BeeCellEditor.jsx

```diff
@@ -0,0 +1,78 @@
+/* global React */
+
+var BeeCell = require('blockly-mooc/src/maze/beeCell');
+
+module.exports = React.createClass({
+  propTypes: {
+    cell: React.PropTypes.object.isRequired,
+    row: React.PropTypes.number.isRequired,
+    col: React.PropTypes.number.isRequired,
+    onUpdate: React.PropTypes.func.isRequired,
+  },
+
+  handleChange: function (event) {
+    var values = this.props.cell.serialize();
+    values[event.target.name] = (event.target.value === 'undefined' || event.target.value === '') ? undefined : parseInt(event.target.value);
```

Comment:
Could right this as `values[event.target.name] = parseInt(event.target.value) || undefined;` since NaN is falsey. I guess that would mean something like "asdf" would also now end up as undefined instead of NaN which may or may not be desired.